File: Week 2/Lab/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    // TODO : Get a sentences from user
    string text = get_string("Text: ");
    float l = 0;
    float w = 1;
    float s = 0;
    int n = strlen(text);
    for (int i = 0; i < n; i++)
    {
        // TODO : Count letter
        if ((text[i] >= 65 && text[i] <= 90) || (text[i] >= 97 && text[i] <= 122))
        {
            l++;
        }
        // TODO : Count word
        if (text[i] == 32)
        {
            w++;
        }
        // TODO : Count sentences
        if (text[i] == 46 || text[i] == 63 || text[i] == 33)
        {
            s++;
        }
    }
    // TODO : Calculate the degree of difficulty
    float L = 100 * (l / w);
    float S = 100 * (s / w);
    int index = round(0.0588 * L - 0.296 * S - 15.8);
    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}
// Smahdimirbagheri

File: Week 6 Python/Problem set 6/sentimental-readability/readability.py
# Import library
import string
from cs50 import get_string
# Give input
txt = get_string("Text: ")
# Count letters
letters = 0
for i in txt:
    if (i.isalpha()):
        letters += 1
# Count words
words = len(txt.split())
# Count sentences
sentences = 0
for j in txt:
    if (j == "." or j == "?" or j == "!"):
        sentences += 1
wordsentences = words / 100
l = letters / wordsentences
s = sentences / wordsentences
index = round(0.0588 * l - 0.296 * s - 15.8)
if index < 1:
    print("Before Grade 1")
elif index >= 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")
# Smahdimirbagheri
